crypto/keys/pem_signer_factory.go

```diff
@@ -17,33 +17,25 @@ package keys
 import (
 	"context"
 	"crypto"
-	"crypto/ecdsa"
-	"crypto/rsa"
-	"crypto/x509"
 	"fmt"
 
 	"github.com/golang/protobuf/ptypes"
 	"github.com/golang/protobuf/ptypes/any"
-	"github.com/google/trillian"
 	"github.com/google/trillian/crypto/keyspb"
 )
 
 // PEMSignerFactory handles PEM-encoded private keys.
-// It supports trees whose PrivateKey field is a:
-// - keyspb.PEMKeyFile
-// - keyspb.PrivateKey
 // It implements keys.SignerFactory.
 type PEMSignerFactory struct{}
 
-// NewSigner returns a crypto.Signer for the given tree.
-func (f PEMSignerFactory) NewSigner(ctx context.Context, tree *trillian.Tree) (crypto.Signer, error) {
-	if tree.GetPrivateKey() == nil {
-		return nil, fmt.Errorf("tree %d has no PrivateKey", tree.GetTreeId())
-	}
-
+// NewSigner uses the information in pb to return a crypto.Signer.
+// pb must be one of the following types:
+// - keyspb.PEMKeyFile
+// - keyspb.PrivateKey
+func (f PEMSignerFactory) NewSigner(ctx context.Context, pb *any.Any) (crypto.Signer, error) {
```

Comment:
We could certainly push the marshaling out to the caller. I'll do this in a follow-up PR.